src/main/java/me/zeroeightsix/fiber/api/schema/type/derived/ConfigTypes.java

```diff
@@ -0,0 +1,276 @@
+            makeNumber(Integer.class, BigDecimal::valueOf, BigDecimal::intValue, Integer.MIN_VALUE, Integer.MAX_VALUE, 1);
+    public static final NumberConfigType<Long> LONG =
+            makeNumber(Long.class, BigDecimal::valueOf, BigDecimal::longValue, Long.MIN_VALUE, Long.MAX_VALUE, 1L);
+    public static final NumberConfigType<Float> FLOAT =
+            makeNumber(Float.class, BigDecimal::valueOf, BigDecimal::floatValue, null, null, null);
+    public static final NumberConfigType<Double> DOUBLE =
+            makeNumber(Double.class, BigDecimal::valueOf, BigDecimal::doubleValue, null, null, null);
+
+    public static <N> NumberConfigType<N> makeNumber(Class<N> actualType, Function<N, BigDecimal> serializer, Function<BigDecimal, N> deserializer, @Nullable N minValue, @Nullable N maxValue, @Nullable N precision) {
+        return new NumberConfigType<>(
+                new DecimalSerializableType(minValue == null ? null : serializer.apply(minValue), maxValue == null ? null : serializer.apply(maxValue), precision == null ? null : serializer.apply(precision)),
+                actualType,
+                deserializer,
+                serializer
+        );
+    }
+
+    /* String-derived types */
+
+    public static final StringConfigType<String> STRING =
+            new StringConfigType<>(StringSerializableType.DEFAULT_STRING, String.class, Function.identity(), Function.identity());
+
+    public static final StringConfigType<Character> CHARACTER =
+            STRING.withMinLength(1).withMaxLength(1).derive(Character.class, s -> s.charAt(0), Object::toString);
+
+    /* Enum-derived types */
+
+    public static <E extends Enum<E>> EnumConfigType<E> makeEnum(Class<E> enumType) {
+        Set<String> validValues = Arrays.stream(enumType.getEnumConstants()).map(Enum::name).collect(Collectors.toSet());
+        return new EnumConfigType<>(new EnumSerializableType(validValues), enumType, e -> Enum.valueOf(enumType, e), Enum::name);
+    }
+
+    /* List-derived types */
+
+    public static <E0, E, U extends ConfigType<E, E0, ?>> ListConfigType<List<E>, E0> makeList(U elementType) {
+        return new ListConfigType<>(
+                new ListSerializableType<>(elementType.getSerializedType()), List.class,
+                l -> {
+                    List<E> ret = new ArrayList<>();
+                    for (E0 e0 : l) {
+                        ret.add(elementType.toRuntimeType(e0));
+                    }
+                    return Collections.unmodifiableList(ret);
+                },
+                l -> {
+                    List<E0> ret = new ArrayList<>();
+                    for (E e : l) {
+                        ret.add(elementType.toPlatformType(e));
+                    }
+                    return Collections.unmodifiableList(ret);
+                }
+        );
+    }
+
+    public static <E0, E, U extends ConfigType<E, E0, ?>> ListConfigType<Set<E>, E0> makeSet(U elementType) {
```

Comment:
`E0` is not a very pretty generic name, maybe use `R` and `S` (from `ConfigType`) instead?